snuba/admin/audit_log/querylog.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum
+from functools import partial
+from typing import Any, Callable, Mapping, MutableMapping, Union
+
+import structlog
+
+from snuba.clickhouse.native import ClickhouseResult
+
+DATETIME_FORMAT = "%B %d, %Y %H:%M:%S %p"
+
+from snuba.admin.audit_log.base import AuditLog
+
+
+class QueryExecutionStatus(Enum):
+    SUCCEEDED = "succeeded"
+    FAILED = "failed"
+
+
+class QuerylogAuditLog(AuditLog):
+    def __init__(self) -> None:
+        self.logger = structlog.get_logger().bind(module=__name__)
+
+    def _record(
+        self, user: str, timestamp: str, action: str, data: Mapping[str, Any]
+    ) -> None:
+        query = data["query"]
+        status = data["status"]
+        start_timestamp = timestamp
+        self.logger.info(
+            event=action,
+            user=user,
+            query=query,
+            status=status.value,
+            start_timestamp=start_timestamp,
+            end_timestamp=datetime.now().strftime(DATETIME_FORMAT),
```

Comment:
Wouldn't the start_timestamp and end_timestamp be the same always in this case? start_timestamp would always be https://github.com/getsentry/snuba/pull/3406/files#diff-fc0f55816e7b2ce9c6703ffa2134728694149604e31e3f4a76ba78494666f90cR21-R22

Reply:
`start_timestamp` is coming from this line, which is executed before the query is run:
https://github.com/getsentry/snuba/pull/3406/files#diff-ba008dc0ca3311bd1841a5ccf1403e9c3bf554f270a1f15b735d4aa8f041a344R67